agent/runner/jobs/mongodb_backup_job.go

```diff
@@ -59,15 +61,17 @@ func NewMongoDBBackupJob(
 	dbConfig DBConnConfig,
 	locationConfig BackupLocationConfig,
 	pitr bool,
+	dataModel backupv1beta1.DataModel,
```

Comment:
We should validate somewhere that we are getting correct value here. For example we can get here `DataModel_DATA_MODEL_INVALID`

Reply:
Added a check to `Run()` to fail if the data model is unsupported.